src/main/host/descriptor/socket/netlink.rs

```diff
@@ -0,0 +1,1215 @@
+        // Send the interface addresses
+        for interface in &common.interfaces {
+            let mut label = Vec::from(interface.label.as_bytes());
+            label.push(0); // Null-terminate
+
+            // List of attribtes sent with the response for the current interface
+            let attrs = [
+                Rtattr::new(None, Ifla::Ifname, Buffer::from(label)).unwrap(),
+                // Not sure about the value of this one, but I always see 1000 from `ip addr`. If
+                // we don't specify this, `ip addr` will create an AF_INET socket and do ioctl. See
+                // https://git.kernel.org/pub/scm/network/iproute2/iproute2.git/tree/ip/ipaddress.c#n168
+                Rtattr::new(None, Ifla::Txqlen, Buffer::from(&1000u32.to_le_bytes()[..])).unwrap(),
+                Rtattr::new(
+                    None,
+                    Ifla::Mtu,
+                    Buffer::from(&interface.mtu.to_le_bytes()[..]),
+                )
+                .unwrap(),
+                // TODO: Add the MAC address through IFLA_ADDRESS and IFLA_BROADCAST
+            ];
+            let flags = if interface.if_type == Arphrd::Loopback {
+                IffFlags::new(&[Iff::Up, Iff::Loopback, Iff::Running])
+            } else {
+                // Not sure about the IFF_MULTICAST, but it's also the one I got from `strace ip addr`
+                IffFlags::new(&[Iff::Up, Iff::Broadcast, Iff::Running, Iff::Multicast])
+            };
+            let ifinfomsg = Ifinfomsg::new(
+                RtAddrFamily::Inet,
+                interface.if_type,
+                interface.index,
+                flags,
+                IffFlags::from_bitmask(0xffffffff), // rtnetlink(7) recommends to set it to all 1s
+                RtBuffer::from_iter(attrs),
+            );
+            let nlmsg = {
+                let len = None;
+                let nl_type = Rtm::Newlink;
+                // The NLM_F_MULTI flag is used to indicate that we will send multiple messages
+                let flags = NlmFFlags::new(&[NlmF::Multi]);
+                // Use the same sequence number as the request
+                let seq = Some(nlmsg.nl_seq);
+                let pid = None;
+                let payload = NlPayload::Payload(ifinfomsg);
+                Nlmsghdr::new(len, nl_type, flags, seq, pid, payload)
+            };
+            nlmsg.to_bytes(&mut buffer).unwrap();
+        }
+        // After sending the messages with the NLM_F_MULTI flag set, we need to send the NLMSG_DONE message
+        let done_msg = {
+            let len = None;
+            let nl_type = Nlmsg::Done;
+            let flags = NlmFFlags::new(&[NlmF::Multi]);
+            // Use the same sequence number as the request
+            let seq = Some(nlmsg.nl_seq);
+            let pid = None;
+            // Linux also emits 4 bytes of zeroes after the header. See `strace ip addr`
+            let payload: NlPayload<Nlmsg, u32> = NlPayload::Payload(0);
```

Comment:
Just a note that if you find a linux source link for line 790, to also add here.